glide/src/androidTest/java/dev/chrisbanes/accompanist/glide/GlideTest.kt

```diff
@@ -0,0 +1,462 @@
+                modifier = Modifier.testTag(GlideTestTags.Image),
+                onRequestCompleted = { latch.countDown() }
+            )
+        }
+
+        // Wait for the onRequestCompleted to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        composeTestRule.onNodeWithTag(GlideTestTags.Image)
+            .assertWidthIsAtLeast(1.dp)
+            .assertHeightIsAtLeast(1.dp)
+            .assertIsDisplayed()
+    }
+
+    @Test
+    fun errorStillHasSize() {
+        val latch = CountDownLatch(1)
+
+        composeTestRule.setContent {
+            GlideImage(
+                data = server.url("/noimage").toString(),
+                modifier = Modifier.preferredSize(128.dp, 128.dp).testTag(GlideTestTags.Image),
+                onRequestCompleted = { latch.countDown() }
+            )
+        }
+
+        // Wait for the onRequestCompleted to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        // Assert that the layout is in the tree and has the correct size
+        composeTestRule.onNodeWithTag(GlideTestTags.Image)
+            .assertIsDisplayed()
+            .assertWidthIsEqualTo(128.dp)
+            .assertHeightIsEqualTo(128.dp)
+    }
+
+    @Test
+    fun content_error() {
+        val latch = CountDownLatch(1)
+        val states = ArrayList<ImageLoadState>()
+
+        composeTestRule.setContent {
+            GlideImage(
+                data = server.url("/noimage").toString(),
+                requestBuilder = {
+                    // Disable memory cache. If the item is in the cache, the fetch is
+                    // synchronous and the dispatcher pause has no effect
+                    skipMemoryCache(true)
+                },
+                modifier = Modifier.preferredSize(128.dp, 128.dp),
+                onRequestCompleted = { latch.countDown() }
+            ) { state ->
+                states.add(state)
+            }
+        }
+
+        // Wait for the onRequestCompleted to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        composeTestRule.runOnIdle {
+            assertThat(states).hasSize(3)
+            assertThat(states[0]).isEqualTo(ImageLoadState.Empty)
+            assertThat(states[1]).isEqualTo(ImageLoadState.Loading)
+            assertThat(states[2]).isInstanceOf(ImageLoadState.Error::class.java)
+        }
+    }
+
+    @Test
+    fun content_success() {
+        val latch = CountDownLatch(1)
+        val states = ArrayList<ImageLoadState>()
+
+        composeTestRule.setContent {
+            GlideImage(
+                data = server.url("/image").toString(),
+                requestBuilder = {
+                    // Disable memory cache. If the item is in the cache, the fetch is
+                    // synchronous and the dispatcher pause has no effect
+                    skipMemoryCache(true)
+                },
+                modifier = Modifier.preferredSize(128.dp, 128.dp),
```

Comment:
If not checking loaded image size, do you need this?

Reply:
Oh, guess it needs a non-zero size to load?

Reply:
Yep exactly